tests/test_calculate_frequencies.py

```diff
@@ -0,0 +1,344 @@
+import os
+import sys
+
+import pandas as pd
+import pytest
+import numpy as np
+
+from api.algorithm_instance import AlgorithmInstance
+import api.recrudescence_utils as recrudescence_utils
+from api.recrudescence_file_parser import RecrudescenceFileParser
+from api.calculate_frequencies import calculate_frequencies3
+
+# NOTE: Makes this reliant on AlgorithmInstance tests passing
+example_file = os.path.join(
+    os.path.dirname(__file__),
+    '../Angola2017_example.xlsx')
+genotypedata, additional = RecrudescenceFileParser.parse_file(example_file)
+
+genotypedata_RR_Benguela = AlgorithmInstance._get_samples_from_site(genotypedata, 'Benguela')
+additional_neutral_Benguela = AlgorithmInstance._replace_sample_names(
+    AlgorithmInstance._get_samples_from_site(additional, 'Benguela'),'Additional_')
+
+genotypedata_RR_Lunda_Sul = AlgorithmInstance._get_samples_from_site(genotypedata, 'Lunda Sul')
+additional_neutral_Lunda_Sul = AlgorithmInstance._replace_sample_names(
+    AlgorithmInstance._get_samples_from_site(additional, 'Lunda Sul'),'Additional_')
+
+genotypedata_RR_Zaire = AlgorithmInstance._get_samples_from_site(genotypedata, 'Zaire')
+additional_neutral_Zaire = AlgorithmInstance._replace_sample_names(
+    AlgorithmInstance._get_samples_from_site(additional, 'Zaire'),'Additional_')
+
+@pytest.fixture
+def Benguela_define_alleles():
+    define_alleles = []
+    columns = ['0', '1']
+    index_0 = pd.DataFrame(
+        [[219.0, 221.0],[221.0, 223.0],[243.0, 245.0],[261.0, 263.0],[223.0, 225.0],
+         [225.0, 227.0],[231.0, 233.0],[237.0, 239.0],[239.0, 241.0],[249.0, 251.0],
+         [217.0, 219.0],[229.0, 231.0],[245.0, 247.0]],
```

Comment:
Holy cow, copying all of these must've taken forever; mad respect. Thanks for doing this!